.circleci/config.yml

```diff
@@ -52,6 +47,12 @@ jobs:
       - store_artifacts:
           path: *TEST_RESULTS_PATH
 
+      # Save go module cache if the go.mod file has changed
+      - save_cache:
```

Comment:
Generally I like to keep this as close to the `restore_cache` as possible so it's easy to see:

1. What cache is restored
2. If the cache isn't there what is run to generate the cache
3. What paths are saved in the cache

The way you have it also works since go fmt downloads dependencies so I will leave it up to you.

Reply:
So in this case would you just leave it where it had been before but reinsert the `run: go mod` line? 

By which I mean how it was done here: https://github.com/hashicorp/go-memdb/blob/9854a2567d0b382dfb6f632cc947f887255e331a/.circleci/config.yml#L21-L32

Reply:
I ask because I like the logical ordering of keeping the caching concerns together so it's easily readable/understandable.

Reply:
Yeah for me I would just do the explicit module download (`go mod download`) and then save it right after. All those go commands after will still try to download the modules first but since it finds it in the modcache it's very fast to just use what it pulled down in a previous `run` step and move on. 

Sorry to push you back and forth on this!

Reply:
:) It's an opportunity for me to think about it and test my opinions to see if I should keep them! I'll take both of these PRs to that format then. Ultimately I think the small efficiency that might be gained here is really just passing around the same bit of margin in different ways.